### PR TITLE
Fix dev next VC service IRI for e2e tests

### PR DESCRIPTION
**Pull request by NSeydoux**
This fixes some issues with the e2e tests, such as the IRI of the VC service for ESS being incorrect. It also extends the coverage of the e2e tests to additional features of the VC library.

# Checklist

- [x] All acceptance criteria are met.
- [x] Relevant documentation, if any, has been written/updated.
- [ ] The changelog has been updated, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: e2e/node/e2e.test.ts [hunk 1]
```diff
@@ -130,33 +144,58 @@ describe.each(serversUnderTest)(
...
     oidcIssuerDisplay,
-    vcIssuerDisplay,
+    vcServiceDisplay,
     vcSubjectDisplay,
     clientId,
     clientSecret
   ) => {
     const oidcIssuer = "https://" + oidcIssuerDisplay;
-    const vcIssuer = "https://" + vcIssuerDisplay;
+    const vcService = "https://" + vcServiceDisplay;
```
Comment (ThisIsMissEm) [inline e2e/node/e2e.test.ts:150]: I'd highly encourage the use of using the `new URL` constructor and enforcing https that way, or via validation of the environment variables.

File: e2e/node/e2e.test.ts [hunk 2]
```diff
@@ -130,33 +141,58 @@ describe.each(serversUnderTest)(
...
+        oidcIssuer,
+        clientId,
+        clientSecret,
       });
+      // The following code snippet doesn't work in Jest, probably because of
+      // https://github.com/standard-things/esm/issues/706 which seems to be
+      // related to https://github.com/facebook/jest/issues/9430. The JSON-LD
+      // module depends on @digitalbazaar/http-client, which uses esm, which
+      // looks like it confuses Jest. Working around this by hard-coding the
+      // endpoints IRIs.
```
Comment (ThisIsMissEm) [inline e2e/node/e2e.test.ts:180]: Can we fork or patch JSON-LD to just use `fetch()`?

Comment (ThisIsMissEm) [reply e2e/node/e2e.test.ts:180]: That said, it looks like ESM is coming to jsonld: https://github.com/digitalbazaar/jsonld.js/issues/399

File: e2e/node/e2e.test.ts (hunk 1 above)
Comment (NSeydoux) [reply e2e/node/e2e.test.ts]: Makes sense. dadaf4419b6dbbd2feed5d83118cac5b489f094e is a minimal effort to have some form of validation, and I opened a ticket to use a dedicated env variables library. One thing we'll have to work around, which is quite annoying, is that GH obfuscates IRIs when displayed in the tests names, which is why we do this horrible hack in the first place.

File: e2e/node/e2e.test.ts (hunk 2 above)
Comment (NSeydoux) [reply e2e/node/e2e.test.ts:180]: I don't think forking or patching should be necessary, if I understand correctly `json-ld.js` supports a custom document loader being provided. I created a ticket to that intent.

**Review by matthieubosquet — APPROVED** [review]
LGTM

**Pull request merged**